pkg/storage/utils/decomposedfs/migrator/migrator.go

```diff
@@ -61,6 +64,82 @@ func New(lu *lookup.Lookup, log *zerolog.Logger) Migrator {
 	}
 }
 
+func (m *Migrator) Migrations() (map[string]MigrationState, error) {
+	err := m.readStates()
+	if err != nil {
+		return nil, err
+	}
+
+	states := map[string]MigrationState{}
+	for _, migration := range allMigrations {
+		if s, ok := m.states[migration]; ok {
+			states[migration] = s
+		} else {
+			states[migration] = MigrationState{
+				State: statePending,
+			}
+		}
+	}
+
+	return states, nil
+}
+
+func (m *Migrator) RunMigration(id string, rollback bool) error {
+	validMigration := false
+	for _, m := range allMigrations {
+		if m == id {
+			validMigration = true
+			break
+		}
+	}
+	if !validMigration {
+		return fmt.Errorf("Invalid migration '%s'", id)
+	}
+
+	lock, err := lockedfile.OpenFile(filepath.Join(m.lu.InternalRoot(), ".migrations.lock"), os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer lock.Close()
+
+	err = m.readStates()
+	if err != nil {
+		return err
+	}
+
+	method := "Up" + id
+	if rollback {
+		method = "Down" + id
+	}
+
+	msg := "Running migration " + id + "..."
+	if rollback {
+		msg += " (down)"
+	}
+	m.log.Info().Msg(msg)
+
+	migrateMethod := reflect.ValueOf(m).MethodByName(method)
+	v := migrateMethod.Call(nil)
```

Comment:
Can we avoid the reflect magic? I'd be more fine with a hardcoded list.

Reply:
@kobergj I came up with https://github.com/cs3org/reva/pull/4083/commits/5076031955383f708b8225b13e75a19c2a155244 which avoids both `reflect` and hard coded lists. wdyt?